### PR TITLE
Make proxy-protocol configurable for the user

### PR DESCRIPTION
**Pull request by webwurst**

### CODE REVIEW

**Comment by webwurst** [comment]
Everything in that file has this format `.Values.configmap "use-proxy-protocol"`, but the global value is `".Values.global.controller.useProxyProtocol"`. Any preference?

**Comment by webwurst** [comment]
I'll give this a try on a test cluster, but feel free to review :)

File: helm/kubernetes-nginx-ingress-controller-chart/templates/configmap.yaml [hunk 1]
```diff
@@ -35,8 +35,14 @@ data:
   server-name-hash-max-size: "{{ index .Values.configmap "server-name-hash-max-size" }}"
   server-tokens: "{{ index .Values.configmap "server-tokens" }}"
   worker-processes: "{{ index .Values.configmap "worker-processes" }}"
+
+  {{- if index .Values.configmap "use-proxy-protocol" }}
+  # Allow user to override value.
+  use-proxy-protocol: "{{ .Values.configmap "use-proxy-protocol" }}"
```
Comment (puja108) [inline helm/kubernetes-nginx-ingress-controller-chart/templates/configmap.yaml]: yeah, I like to use the upstream conform naming of the key here

Comment (pipo02mix) [reply helm/kubernetes-nginx-ingress-controller-chart/templates/configmap.yaml]: But dont you need to define the variable on the [chart values](https://github.com/giantswarm/kubernetes-nginx-ingress-controller/blob/master/helm/kubernetes-nginx-ingress-controller-chart/values.yaml#L10-L24) ?

**Comment by rossf7** [comment]
Tested successfully on `geckon` and `ginger`.

**Pull request merged**